typedb/cluster/database.py

```diff
@@ -142,3 +147,99 @@ def __hash__(self):
 
             def __str__(self):
                 return "%s/%s" % (self._address, self._database)
+
+
+# This class has to live here because of circular class creation between ClusterDatabase and FailsafeTask
```

Comment:
This is highly unpleasant but I can't find any workaround that is any better. Because an instance method of `FailsafeTask` creates a `ClusterDatabase`, and an instance method of `ClusterDatabase` creates a `FailsafeTask`, it is physically impossible to decouple the two in Python. (This may also be the case in NodeJS)